### PR TITLE
[WIP] Refractor the check-for-build method and tests

### PR DESCRIPTION
**Pull request by zelinh**
### Description
I found out a problem regarding of the PR just merged #1628. When we disable `propagate` when trigger the job, the following 
```
echo "Build succeeded, uploading build SHA for that job"
buildUploadManifestSHA(
    inputManifest: "manifests/${INPUT_MANIFEST}",
    jobName: "${TARGET_JOB_NAME}"
)
``` 
will be executed after triggered job failed which not what we expected. We shouldn't upload it if the distribution-build failed. 

The use of try-catch block will avoid this situation, if the `distribution-build` failed, nothing below the job call in the try block will be executed. 
 
### Issues Resolved
continue on #1627 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
Should this be closed in favor of https://github.com/opensearch-project/opensearch-build/pull/1628?

**Review by peternied — CHANGES_REQUESTED** [review]
I think that miss on he previous PR could have been caught by unit tests.  Before we do more work here can we add tests for this jenkinsfile that cover the success and failure build cases?

Note; we should revert the previous PR if this is impacting the builds negatively

**Comment by zelinh** [comment]
> > I think that miss on he previous PR could have been caught by unit tests. Before we do more work here can we add tests for this jenkinsfile that cover the success and failure build cases?
> > Note; we should revert the previous PR if this is impacting the builds negatively
> 
> Oh my bad, read too quickly. Sounds like the SHA should be uploaded inside the `build` job instead?

Yes, that's another way I was considering.

**Comment by zelinh** [comment]
> I think that miss on he previous PR could have been caught by unit tests. Before we do more work here can we add tests for this jenkinsfile that cover the success and failure build cases?
> 
> Note; we should revert the previous PR if this is impacting the builds negatively

Yes, adding a test for `check-for-build` is definitely needed as we have a meta issue here. #1468; while that's not in the scope of this PR. This PR provides an approach for resolving #1627 and moving SHA upload to build could also be another approach but it will require to be added in both `distribution-build-opensearch` & `distribution-build-opensearch-dashboards`. WDYT?  @dblock

**Comment by dblock** [comment]
> > I think that miss on he previous PR could have been caught by unit tests. Before we do more work here can we add tests for this jenkinsfile that cover the success and failure build cases?
> > Note; we should revert the previous PR if this is impacting the builds negatively
> 
> Yes, adding a test for `check-for-build` is definitely needed as we have a meta issue here. #1468; while that's not in the scope of this PR. This PR provides an approach for resolving #1627 and moving SHA upload to build could also be another approach but it will require to be added in both `distribution-build-opensearch` & `distribution-build-opensearch-dashboards`. WDYT? @dblock

You can't use try/catch because it will swallow the error from uploading the SHA, and we'll never know.

File: jenkins/check-for-build.jenkinsfile [hunk 1]
```diff
@@ -76,11 +76,6 @@ pipeline {
                                 string(name: 'INPUT_MANIFEST', value: "${INPUT_MANIFEST}"),
                                 string(name: 'TEST_MANIFEST', value: "${TEST_MANIFEST}")
                             ], wait: true, propagate: false
-                            echo "Build succeeded, uploading build SHA for that job"
```
Comment (peternied) [inline jenkins/check-for-build.jenkinsfile:79]: We should still add tests to confirm the behavior is correct

**Review by peternied — CHANGES_REQUESTED** [review]

**Comment by zelinh** [comment]
Closing this for now since this is not being worked actively.

**Pull request closed without merging**